src/main/java/at/ac/tuwien/kr/alpha/solver/ArrayAssignment.java

```diff
@@ -361,6 +362,15 @@ public Entry get(int atom) {
 		return assignment.get(atom);
 	}
 
+	Integer getSomeMBTAssignedAtom(Grounder grounder) {
```

Comment:
This is giving me a headache. We now have `Grounder#getAtom` which just forwards to the grounder's `AtomStore`, and then the solver calls `ArrayAssignment#getSomeMBTAssignedAtom` passing in the grounder. Additionally, this method depends on a concrete implementation of `Atom`. Very odd.

It seems that both grounder and solver need to be aware of the assignment. If we would have this then the solver could just ask the grounder for MBT assigned atoms under an assignment. We would have something like: `Integer Grounder#getSomeMBTAssignmentAtom(Assignment)` (cf. `Grounder#getUnassignedAtoms`). That would seem a little more natural to me, and `BasicAtom` would not leak into the assignment (see new import in this file). Maybe I am missing something and this is not possible after all...

Reply:
What we need here is an MBT-assigned, ordinary atom. Unfortunately, the assignment does not know which atom id represents an ordinary atom, this knowledge is currently only in the grounder. Hence the `getAtom` and the test for `BasicAtom` to decide if the atom is an ordinary one.

Reply:
I opened #132 to follow up on this.